model-view-viewmodel/README.md

```diff
@@ -0,0 +1,227 @@
+            <listhead>
+                <listheader label="Book Name"/>
+                <listheader label="Author"/>               
+            </listhead>
+            <template name="model" var="book">
+                <listitem >
+                    <listcell label="@bind(book.name)"/>
+                    <listcell label="@bind(book.author)"/>
+                </listitem>
+            </template>
+        </listbox>
+    </vbox>
+    <toolbar>
+        <button label="Delete" onClick="@command('deleteBook')" disabled="@load(empty vm.selectedBook)" />
+    </toolbar>
+    <hbox style="margin-top:20px" visible="@bind(not empty vm.selectedBook)">
+		<vbox>
+			<hlayout>
+				Book Name : <label value="@bind(vm.selectedBook.name)" style="font-weight:bold"/>
+			</hlayout>
+			<hlayout>
+				Book Author : <label value="@bind(vm.selectedBook.author)" style="font-weight:bold"/>
+			</hlayout>
+			<hlayout>
+				Book Description : <label value="@bind(vm.selectedBook.description)" style="font-weight:bold"/>
+			</hlayout>
+		</vbox>
+	</hbox>
+</window>
+</zk>
+```
+
+## Class diagram
+
+![alt text](./etc/model-view-viewmodel.png "MVVM pattern class diagram")
+
+## Applicability
+
+* When looking for clean architecture, with better reusability, testability and maintainability.
+
+## Tutorials
+
+* [Zkoss Demo](https://www.zkoss.org/zkdemo/getting_started/mvvm)
+* [Learn MVVM](https://www.learnmvvm.com/)
+* [Android Developer CodeLabs](https://codelabs.developers.google.com/codelabs/android-databinding)
+
+## Typical Use Case
+
+* Android Apps
+* .Net Frameworks
+* JavaScript Frameworks
+
+## Real world examples
+
+* ZK Framework [zkoss.org](https://www.zkoss.org/)
+* KnockoutJS [knockoutjs.com](https://knockoutjs.com/)
+
+## Consequences
+
+* John Gossman has criticized the MVVM pattern and its application in specific uses, stating that MVVM can be "overkill" when creating simple user interfaces. For larger applications, he believes that generalizing the viewmodel upfront can be difficult, and that large-scale data binding can lead to lower performance - Ref: [MVVM-Wiki](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel)
+
+* Can be hard to design ViewModel for larger applications.
+* For complex databinding, debugging can be difficult.
+
+## Credits
+
+* [ZK MVVM](https://www.zkoss.org/wiki/ZK%20Developer's%20Reference/MVVM)
+* [GeeksforGeeks  MVVM Intro](https://www.geeksforgeeks.org/introduction-to-model-view-view-model-mvvm/)
+* [ZK MVVM Book](http://books.zkoss.org/zk-mvvm-book/9.5/)
+* [Microsoft MVVM](https://docs.microsoft.com/en-us/archive/msdn-magazine/2009/february/patterns-wpf-apps-with-the-model-view-viewmodel-design-pattern)
+
+Note: 
+* To deploy this, go to model-view-viewmodel folder and run: 
+* mvn clean install
+* mvn jetty:run -Djetty.http.port=9911
+* In browser, http://localhost:9911/model-view-viewmodel/
```

Comment:
Maybe these instructions should be placed next to the programmatic example?

Reply:
done